dependency.lic

```diff
@@ -772,257 +773,467 @@ class SlackbotManager
 end
 
 class SetupFiles
-  def initialize(debug)
+  # http://www.druby.org/sidruby/5-3-thread-safe-communication-using-locking-mutex-and-monitormixin.html#sec.monitormixin
+  include MonitorMixin
+
+  class FileInfo
+    attr_reader :path, :name, :mtime
+
+    def initialize(path:, name:, data:, mtime:)
+      @path = path
+      @name = name
+      @data = data
+      @mtime = mtime
+    end
+
+    def data
+      # Create deep clone of data to mitigate scripts changing settings in-memory
+      # because that leads to hard to track down bugs and unpredictable script behaviors.
+      Marshal.load(Marshal.dump(@data))
```

Comment:
Nice for for safety, although we should also make sure we're correcting any submissions that are modifying settings.

Reply:
Can you clarify what "make sure we're correcting any submissions that are modifying settings"? Do you mean other humans who submit code changes to this file or something else? Thanks

Reply:
Just an idle thought that we should be keeping an eye open for script submissions that are modifying the settings themselves.